### PR TITLE
feat(opensearch): add OpenSearch 2.13 engine version

### PR DESCRIPTION
**Pull request by timbru31**
### Reason for this change

[Amazon OpenSearch Service now supports OpenSearch version 2.13](https://aws.amazon.com/about-aws/whats-new/2024/05/amazon-opensearch-supports-opensearch-version-2-13/)

### Description of changes

Adds support for OpenSearch engine version 2.13

### Description of how you validated changes

`yarn test`

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by timbru31** [comment]
Clarification Request

Other PRs did not change the README either when the new engine versions were being added.

**Comment by timbru31** [comment]
Exemption Request

Other PRs did not change the README either when the new engine versions were being added.

**Comment by TheRealAmazonKendra** [comment]
Excuse me pr bot, an exemption request was added here. Label it! (This is me testing my fix)

**Comment by TheRealAmazonKendra** [comment]
Good PR Bot

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
Thank you for your contribution! Just a couple of comments:

Integration test changes shouldn't remove coverage from older versions, we still need those tested against as long as customers can still use them.

Also, are all configuration properties valid for 2.13 or are there limitations? If not, we will need to validate for them in code and add those limitation to the README.

File: packages/aws-cdk-lib/aws-opensearchservice/lib/version.ts [hunk 1]
```diff
@@ -90,15 +90,18 @@ export class EngineVersion {
...
+
   /**
-   * AWS OpenSearch 2.10
+   * AWS OpenSearch 2.11
    * @deprecated use latest version of the OpenSearch engine
    **/
-  public static readonly OPENSEARCH_2_10 = EngineVersion.openSearch('2.10');
-
-  /** AWS OpenSearch 2.11 */
   public static readonly OPENSEARCH_2_11 = EngineVersion.openSearch('2.11');
```
[the remark this answers is not in the document]
Comment (timbru31) [reply packages/aws-cdk-lib/aws-opensearchservice/lib/version.ts]: Thanks for the documentation hint, I've updated the changes accordingly.

**Comment by timbru31** [comment]
> Thank you for your contribution! Just a couple of comments:
> 
> Integration test changes shouldn't remove coverage from older versions, we still need those tested against as long as customers can still use them.
> 
> Also, are all configuration properties valid for 2.13 or are there limitations? If not, we will need to validate for them in code and add those limitation to the README.


Hi Kendra,
I've updated the version in the integration tests as the previous PR (https://github.com/aws/aws-cdk/pull/28109/files#diff-3f816d35b99012305de1195d096d4315ed0058906a20f09202b742a73b3e4bcdR25) also did. If you'd like I can leave them untouched and only update packages/aws-cdk-lib/aws-opensearchservice/test/domain.test.ts - please let me know.

Also I am not aware of any changes in 2.13 that should cause issues with the properties.

**Comment by TheRealAmazonKendra** [comment]
> > Thank you for your contribution! Just a couple of comments:
> > Integration test changes shouldn't remove coverage from older versions, we still need those tested against as long as customers can still use them.
> > Also, are all configuration properties valid for 2.13 or are there limitations? If not, we will need to validate for them in code and add those limitation to the README.
> 
> Hi Kendra, I've updated the version in the integration tests as the previous PR (https://github.com/aws/aws-cdk/pull/28109/files#diff-3f816d35b99012305de1195d096d4315ed0058906a20f09202b742a73b3e4bcdR25) also did. If you'd like I can leave them untouched and only update `packages/aws-cdk-lib/aws-opensearchservice/test/domain.test.ts` - please let me know.
> 
> Also I am not aware of any changes in 2.13 that should cause issues with the properties.

You definitely made a fair assumption that we'd want the same thing again here. I typically prefer to see additional integ tests added instead of updating the old ones, but I'll let this one go in this case (watch this very comment come back to bite me at some point, lol.

**Review by TheRealAmazonKendra — APPROVED** [review]

**Pull request merged**